plugins/experimental/remap_stats/remap_stats.cc

```diff
@@ -141,11 +141,11 @@ handle_post_remap(TSCont cont, TSEvent event ATS_UNUSED, void *edata)
 }
 
 void
-create_stat_name(ts::FixedBufferWriter &stat_name, std::string_view const h, std::string_view const b)
+create_stat_name(swoc::FixedBufferWriter &stat_name, std::string_view const h, std::string_view const b)
 {
-  stat_name.reset().clip(1);
+  stat_name.clear().restrict(1);
```

Comment:
I know this is existing code and this is just changing to the swoc names, but what does this restrict/restore do?

Reply:
It adjusts the capacity temporarily. `restrict(1)` decreases the capacity by one, so that there is a byte in reserve. This is useful when specific terminal characters are required (in this case, I think it's a terminal nul). The intermediate data can be written, and the the capacity `restore`'d so the terminal nul can always be written to the buffer.

Reply:
oooh, its restrict BY, not restrict TO.  I see. thanks